Userland/Libraries/LibC/termios.cpp

```diff
@@ -51,79 +51,20 @@ speed_t cfgetospeed(const struct termios* tp)
     return tp->c_ospeed;
 }
 
-static int baud_rate_from_speed(speed_t speed)
-{
-    int rate = -EINVAL;
-    switch (speed) {
-    case B0:
-        rate = 0;
-        break;
-    case B50:
-        rate = 50;
-        break;
-    case B75:
-        rate = 75;
-        break;
-    case B110:
-        rate = 110;
-        break;
-    case B134:
-        rate = 134;
-        break;
-    case B150:
-        rate = 150;
-        break;
-    case B200:
-        rate = 200;
-        break;
-    case B300:
-        rate = 300;
-        break;
-    case B600:
-        rate = 600;
-        break;
-    case B1200:
-        rate = 1200;
-        break;
-    case B1800:
-        rate = 1800;
-        break;
-    case B2400:
-        rate = 2400;
-        break;
-    case B4800:
-        rate = 4800;
-        break;
-    case B9600:
-        rate = 9600;
-        break;
-    case B19200:
-        rate = 19200;
-        break;
-    case B38400:
-        rate = 38400;
-        break;
-    }
-
-    return rate;
-}
-
+// We do not have to check for validity in the cfset{i,o}speed functions.
+// POSIX says it's okay if we only fail after calling tcsetattr.
 int cfsetispeed(struct termios* tp, speed_t speed)
 {
-    auto ispeed = baud_rate_from_speed(speed);
-    if (ispeed > 0) {
-        tp->c_ispeed = ispeed;
-    }
-    __RETURN_WITH_ERRNO(ispeed, 0, -1);
+    tp->c_ispeed = speed;
+    errno = 0;
```

Comment:
POSIX also says that we should not clear `errno` on success, so I think this is wrong.